push to insecure registry (`panic: runtime error: index out of range [0] with length 0`)

I would like to use rootless buildkit in a Kubernetes Job and trying to use the provided rootless template from the [examples folder](../blob/master/examples/kubernetes/job.rootless.yaml).
Build works greate, but pushing fails with:
```
#10 ERROR: failed to do request: Head "https://registry-svc:5000/v2/foo/blobs/sha256:0443cbc5100eb115b6787f6233e4b00a731a6609c62a07a509bdb43083e61f55": http: server gave HTTP response to HTTPS client
------
 > importing cache manifest from registry-svc:5000/foo:latest:
------
------
 > exporting to image:
------
error: failed to solve: failed to do request: Head "https://registry-svc:5000/v2/foo/blobs/sha256:0443cbc5100eb115b6787f6233e4b00a731a6609c62a07a509bdb43083e61f55": http: server gave HTTP response to HTTPS client
```

I then stumbled upon #2044 and used the mentioned approach mounting a `buildkit.toml` file. My `.toml` file looks like this:
```toml
debug = false
[worker.containerd]
  namespace = "buildkit"
[registry."registry-svc:5000"]
  http = true
  insecure = true
```
It is mounted under `/home/user/.config/buildkit/buildkit.toml`.

However this results in the following error:
```
could not connect to unix:///run/user/1000/buildkit/buildkitd.sock after 10 trials
========== log ==========
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.main()
	/src/cmd/buildkitd/main.go:167 +0x13cc
[rootlesskit:child ] error: command [buildkitd --oci-worker-no-process-sandbox --addr=unix:///run/user/1000/buildkit/buildkitd.sock] exited: exit status 2
[rootlesskit:parent] error: child exited: exit status 2
sh: can't kill pid 9: No such process
```

Any idea what I might do wrong?

--- a/cmd/buildkitd/main.go
+++ b/cmd/buildkitd/main.go
@@ -375,7 +375,7 @@ func defaultConf() (config.Config, error) {
 		if !errors.As(err, &pe) {
 			return config.Config{}, err
 		}
-		return cfg, nil
+		logrus.Warnf("failed to load default config: %v", err)
 	}
 	setDefaultConfig(&cfg)
 

